### PR TITLE
Added entity/metadata model as per our discussions today.

### PR DESCRIPTION
**Pull request by jacarey**
Basic data access layer added.

### CODE REVIEW

File: src/main/scala/org/broadinstitute/dsde/agora/server/model/AgoraEntityModel.scala [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+package org.broadinstitute.dsde.agora.server.model
+
+import java.util.Date
+
+case class AgoraMetadata(id: Option[Integer] = None, namespace: String, name: String, synopsis: String,
```
Comment (geoffjentry) [inline src/main/scala/org/broadinstitute/dsde/agora/server/model/AgoraEntityModel.scala]: I get why you ordered this the way you did but putting defaulted params before non-defaulted params can lead to pain & anguish

File: src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  }
+
+  def insert(entity: AgoraEntity): AgoraEntity = {
+    val entityDbObject = grater[AgoraEntity].asDBObject(entity)
+    collection.insert(entityDbObject)
+    find(entityDbObject).head
+  }
+
+  def find(query: Imports.DBObject) = {
+    (for (entity <- collection.find(query)) yield grater[AgoraEntity].asObject(entity)).toList
```
Comment (geoffjentry) [inline src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala]: If there's not a particular reason to use List (perhaps there is?) use Vector. It's way mo bettah

File: src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import org.broadinstitute.dsde.agora.server.dataaccess.AgoraDao
+import org.broadinstitute.dsde.agora.server.model.AgoraEntity
+
+class AgoraMongoDao(collection: MongoCollection) extends AgoraDao {
+
+  def findPayloadByRegex(regex: String): Seq[AgoraEntity] = {
+    find("payload" $regex regex)
+  }
+
+  def findByName(name: String): Seq[AgoraEntity] = {
```
Comment (bradtaylor) [inline src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala]: Will we want to do a find by name/namespace?

Also, for the present story, do we want findById?

Comment (jacarey) [inline src/main/scala/org/broadinstitute/dsde/agora/server/dataaccess/mongo/AgoraMongoDao.scala]: Both added in the next PR.

**Comment by bradtaylor** [comment]
Looks great to me, taking into account Jeff's changes

**Pull request merged**